spec/nightwing/sidekiq/queue_stats_spec.rb

```diff
@@ -1,40 +1,69 @@
 require "spec_helper"
 
 describe Nightwing::Sidekiq::QueueStats do
-  subject { Nightwing::Sidekiq::QueueStats.new(client: Nightwing::DebugClient.new) }
+  subject { Nightwing::Sidekiq::QueueStats.new(client: Nightwing::DebugClient.new, disabled_metrics: disabled_metrics) }
+  let(:disabled_metrics) { [] }
 
   describe "#call" do
-    let(:fake_queue) { Struct.new(:size, :latency).new(0, 0) }
+    let(:fake_queue) { instance_double(::Sidekiq::Queue, name: "fake_queue", size: 0, latency: 0) }
 
     before do
-      allow(Sidekiq::Queue).to receive(:new).and_return fake_queue
+      allow(Sidekiq::Queue).to receive(:new).with(fake_queue.name).and_return fake_queue
     end
 
     context "when everything just works" do
       it "increments process count" do
-        expect(subject.client).to receive(:measure).with("sidekiq.default.size", 0).and_call_original
-        expect(subject.client).to receive(:measure).with("sidekiq.default.latency", 0).and_call_original
-        expect(subject.client).to receive(:increment).with("sidekiq.default.processed").and_call_original
+        expect(subject.client).to receive(:measure).with("sidekiq.fake_queue.size", 0).and_call_original
+        expect(subject.client).to receive(:measure).with("sidekiq.fake_queue.latency", 0).and_call_original
+        expect(subject.client).to receive(:increment).with("sidekiq.fake_queue.processed").and_call_original
 
-        subject.call(nil, nil, "default") do
+        subject.call(nil, nil, fake_queue.name) do
           # beep
         end
       end
+
+      context "and queue depth metrics are disabled" do
+        let(:disabled_metrics) { [:queue_depth] }
+
+        it "does not report them" do
+          expect(subject.client).not_to receive(:measure).with("sidekiq.fake_queue.size", any_args).and_call_original
```

Comment:
As we're asserting `not_to receive`, is the `and_call_original` needed here (and in the context below)?